src/argman.rs

```diff
@@ -0,0 +1,712 @@
+
+use std::collections::HashMap;
+use std::env;
+
+#[derive(Debug)]
+enum ArgType {
+    ArgBool,
+    ArgMultistr,
+    ArgMapStr,
+    ArgStr,
+}
+
+#[derive(Debug)]
+struct ArgumentHelp {
+    description: String,
+    arg_type: ArgType,
+    default: Option<String>,
```

Comment:
Seems like a good place to use an enum?

Reply:
I actually tried that first.
But then I was trapped in errors I didn't always understand or just APIs for ArgMan that sucked even more than this one.
Can I try that in a different PR just for my own sanity?

Reply:
I'd encourage instead to rethink the whole API as a more simplified rust-lightning-bitcoinrpc-specific thing that uses enums instead of lots of maps and checking.